test/controllers/vatLodgingOfficer/OfficerContactDetailsControllerSpec.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.vatLodgingOfficer
+
+import fixtures.VatRegistrationFixture
+import helpers.{S4LMockSugar, VatRegSpec}
+import models.S4LKey
+import models.view.vatLodgingOfficer.OfficerContactDetails
+import models.view.vatTradingDetails.vatChoice.VoluntaryRegistration
+import org.mockito.Matchers
+import org.mockito.Matchers.any
+import org.mockito.Mockito._
+import play.api.libs.json.Json
+import play.api.test.FakeRequest
+import uk.gov.hmrc.http.cache.client.CacheMap
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class OfficerContactDetailsControllerSpec extends VatRegSpec with VatRegistrationFixture with S4LMockSugar {
+
+  import cats.instances.future._
+  import cats.syntax.applicative._
+
+  object TestOfficerContactDetailsController extends OfficerContactDetailsController(ds)(mockS4LService, mockVatRegistrationService) {
+    override val authConnector = mockAuthConnector
+  }
+
+  val fakeRequest = FakeRequest(controllers.vatLodgingOfficer.routes.OfficerContactDetailsController.show())
+
+  s"GET ${controllers.vatLodgingOfficer.routes.OfficerContactDetailsController.show()}" should {
+
+    "return HTML when there's nothing in S4L and vatScheme contains data" in {
+      when(mockS4LService.fetchAndGet[OfficerContactDetails]()(any(), any(), any()))
+        .thenReturn(Future.successful(None))
+      when(mockVatRegistrationService.getVatScheme()(any[HeaderCarrier]()))
+        .thenReturn(Future.successful(validVatScheme))
+
+      callAuthorised(TestOfficerContactDetailsController.show()) {
+        _ includesText "What are your contact details?"
+      }
+    }
+
+
+    "return HTML when there's an answer in S4L" in {
+      when(mockS4LService.fetchAndGet[OfficerContactDetails]()
+        (Matchers.eq(S4LKey[OfficerContactDetails]), any(), any()))
+        .thenReturn(Future.successful(Some(validOfficerContactDetails)))
+
+      callAuthorised(TestOfficerContactDetailsController.show) {
+        _ includesText "What are your contact details?"
+      }
+    }
+
+    "return HTML when there's nothing in S4L and vatScheme contains no data" in {
+      when(mockS4LService.fetchAndGet[OfficerContactDetails]()(Matchers.eq(S4LKey[OfficerContactDetails]), any(), any()))
```

Comment:
in this same file you use the s4l mock sugar `save4LaterReturns` and also you use this mockito mocking, which is inconsistent